genai-perf/genai_perf/config/input/config_endpoint.py

```diff
@@ -180,48 +183,44 @@ def _check_for_valid_url(self, url: str) -> None:
                 "(e.g., ':8002')."
             )
 
-    def _check_dynamic_grpc(self) -> None:
-        if self.service_kind == "dynamic_grpc" and not self.grpc_method:
-            raise ValueError(
-                "User Config: service_kind 'dynamic_grpc' requires a grpc_method to be specified"
-            )
-        elif self.service_kind != "dynamic_grpc" and self.grpc_method:
-            raise ValueError(
-                "User Config: grpc_method is only valid with service_kind 'dynamic_grpc'"
-            )
-
     ###########################################################################
     # Infer Methods
     ###########################################################################
     def infer_settings(self, model_name: str) -> None:
         # IMPORTANT: There is a nasty dependency ordering between the infer methods
         # For readability they are broken up into separate methods, that daisy chain
         # their calls
-        self.infer_type(model_name)
+        self.infer_service_kind(model_name)
 
-    def infer_type(self, model_name: str) -> None:
-        if self.service_kind == "openai" and not self.get_field("type").is_set_by_user:
-            raise ValueError(
-                "User Config: type must be set when service_kind is 'openai'"
+    def infer_service_kind(self, model_name: str) -> None:
+        if self.get_field("service_kind").is_set_by_user:
+            logger.warning(
+                "User Config: service_kind is deprecated. The service kind "
+                "will be inferred from the endpoint type, grpc_method, or "
+                "backend."
             )
 
+        if self.type:
+            endpoint_config = endpoint_type_map[self.type]
+            self.service_kind = endpoint_config.service_kind
+        elif self.get_field("grpc_method").is_set_by_user:
+            self.service_kind = "dynamic_grpc"
+        else:
+            # Default case
+            self.service_kind = "triton"
+
+        self.infer_type(model_name)
```

Comment:
With the removal of user setting `--service-kind`, would it be correct to say that the default `--endpoint-type` is `kserve` instead of `None`? I know it's not literally `default="kserve"` in the argparse code, but I think by the time it gets here and sets default service_kind = "triton" and then `infer_type -> self.type = "kserve"`, it works out that way, right?

Reply:
That's correct. We're removing the defaults in argparse very soon.

Reply:
> That's correct. We're removing the defaults in argparse in [a separate PR](https://nvidia.slack.com/archives/C07FW51CE1J/p1744302323474549).

I think this link is self-pointing at the slack link back to this PR, is that the correct link?

Reply:
Whoops, thanks for pointing that out. Here is the new link: https://github.com/triton-inference-server/perf_analyzer/pull/350/files#diff-89352fdfacaf2688a3d610f2686415e446f1d2a916d705ef20612f385e416b90

Based on Ryan and Brian's feedback, I went ahead and just removed the inference of dynamic_grpc from grpc_method, which was making things a bit wonkier. It means that those using dynamic_grpc need to specify it as an endpoint, but I think that's the right choice to make everything consistent.

So now the defaults are Triton and kserve. If a user specifies a different endpoint-type (e.g. dynamic_grpc, completions, etc.), that will be used instead. This seems more consistent and maintainable.